src/segment/reactor.rs

```diff
@@ -49,6 +50,15 @@ impl Reactor {
         let (event, cap_guard) = receiver.recv().await.expect("sender closed, processor exit");
         match event {
             Incoming::AppendEvent(pending_event) => {
+                if pending_event.data.len() > PendingEvent::MAX_WRITE_SIZE {
+                    let mut large_event_writer =
+                        LargeEventWriter::new(selector.delegation_token_provider.clone());
+                    large_event_writer
+                        .write(factory, selector, pending_event)
+                        .await
+                        .map_err(|_| "large event writer failed to write")?;
+                    return Ok(());
```

Comment:
To preserve order we need to make sure to first flush the prior events.

Reply:
As that is in a caller we can't be sure it was actually done at this point in the code. So here there should either be an assertion to check that it was done by the caller, or it should be done here.

Reply:
`self.flush().await` can make sure that all the pior events has been acked and then send the large event to the reactor task to proceed. In the reactor task, do you think we should check the pending queue size again? if it is larger then zero, make an assert?

Reply:
Yes. That sounds good.

Reply:
done